dask_kubernetes/core.py

```diff
@@ -393,6 +396,73 @@ def __init__(
         self.kwargs = kwargs
         super().__init__(**self.kwargs)
 
+    def _clean_pod_templates(self):
+        for pod_template, pod_type in [
+            (self.pod_template, "worker"),
+            (self.scheduler_pod_template, "scheduler"),
+        ]:
```

Comment:
I wonder if you could call this function twice and pass these in as args. Would make it more readable.

Reply:
Yeah I see what you mean ! I'll make the modifications.

Reply:
The thing though is that there is a bit of logic to assign the scheduler template to the pod template by default. And this requires having knowledge of both templates. I can move this part out of the function, but I fell it belongs more there than in the main `_start` method.

Reply:
I think my preference would be to have it in the main `_start` function. It makes the start function clearer to read.

Reply:
Ok I see what you mean. I'll change it.